### ISSUE TITLE
Crash if google play is not installed

### ISSUE THREAD

**Issue by MisterGitHubber** [CypherpunkArmory/UserLAnd#298]
After installing system I click on it and got crash. I see logs - ActivityNotFoundException. Surround this code in try/catch for catching ActivityNotFoundException please

**Comment by AdamMatthewLTM** [comment]
@MisterGitHubber ,

Thank you for your post. Can you please reply here with a detailed step-by-step procedure I can follow to replicate this matter as per the template provided to users filing issues?

Thank you for your time and patience in these matters,
-Adam

**Comment by MisterGitHubber** [comment]
@AdamMatthewLTM, just surround your code for launching ConnectBot in Google Play with try/catch. google play may be is not installed and user can't know why UserLAnd crashed!

final String appPackageName = getPackageName(); // getPackageName() from Context or Activity object

    try {
        startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=" + appPackageName)));
} catch (ActivityNotFoundException anfe) {
    // Any another link for downloading ConnectBot
}

**Comment by l3laze** [comment]
@MisterGitHubber --

* Which version of Android, or a custom Android OS, are you using?
* If Google Play/Google Play Services **are installed and are disabled**, and you're okay with temporarily allowing them, does it work as expected with them enabled?
* If GP/GP Services **are installed and are not disabled**, have you tried updating them?


----


Provided the issue is caused by what OP says:


Google Play/GP Services should only be unavailable on rooted devices using a custom OS; or devices which the owner has deleted the Google Play app and/or Google Play Services; or on any device (root or not) where GP/GP Services have been disabled. It could theoretically also be caused by running an old version of either, but that's only theory.


[Android-x86](http://www.android-x86.org) is rooted by default and IIRC *does not* include GP/GPS, but I don't currently have a PC to confirm that, and won't for a long while. It would of course be possible to disable them if they are installed though, via Settings -> Apps -> Whatever App -> Disable. It is necessary to uninstall updates before an app can be disabled though as it uses the "Uninstall Updates" button.


This should make it pretty easy to recreate like so:

* [Create a VM](http://www.android-x86.org/documents/virtualboxhowto).
* After it starts, install UserLAnd & ConnectBot and start to confirm whether they're working.
* If they fail and Google Play/Google Play Services are not installed the error log should be similar to OP's.
* If it succeeds and GP/GP Services are installed, disable them and try to run UserLAnd + ConnectBot again. It should now crash as it did for OP.


----


Looking through the source (which I can't really understand) it seemed like maybe the file ServerService.kt's function startSshClient needs it's Intent.type set as with startVncClient, but I can barely read Java, let alone Kotlin or Android, and understand very little beyond the basics so that's just a perceived possible issue from a noob's POV.

**Comment by MatthewTighe** [comment]
This will be fixed in the next release. 

FYI, we're adding a feature for single-click launch of popular and useful apps (such as Firefox) onto a pregenerated filesystem. Since nothing is stopping people from manually installing things like that and this is a feature entirely of convenience, it will be our first paid feature.  The only payment method we will support at release is through Google Play Services, so you won't have access to the feature until we support additional payment methods or you install Google Play.

**Comment by MisterGitHubber** [comment]
Yes I use custom OS without GP. But I don't understand why UserLAnd crashed when I click on installed Linux. Nothing. Just crash. Where information about non-installed ConnectBot? Before trying launching another app user must be know that will be launched now.
